pkg/codegen/go/gen.go

```diff
@@ -2594,35 +2591,80 @@ func generatePackageContextMap(tool string, pkg *schema.Package, goInfo GoPackag
 	// compute set of names generated by a type
 	// handling any potential collisions via remapping along the way
 	scanType := func(t schema.Type) {
+		getNames := func(name, suffix string) []string {
+			names := []string{}
+			names = append(names, name+suffix)
+			names = append(names, name+suffix+"Input")
+			names = append(names, name+suffix+"Output")
+			return names
+		}
+
 		switch t := t.(type) {
 		case *schema.ObjectType:
 			pkg := getPkgFromToken(t.Token)
-			// maintain support for duplicate tokens for types and resources in Kubernetes
+			// maintain support for duplicate tokens for types and resources
 			if resSeen[t.Token] {
 				pkg.duplicateTokens[t.Token] = true
-				return
```

Comment:
Intentional right?

Reply:
yes! without this the type names are never processed and the enum never sees the collision